components/catalogs/front/src/components/ProductSelection/ProductSelection.integration.tsx

```diff
@@ -128,3 +130,51 @@ test('it updates the state when a criterion is removed', async () => {
 
     expect(onChange).toHaveBeenCalledWith({});
 });
+
+test('it shows a warning and lock the add button when the criteria limit is reached', async () => {
+    const criteria = [];
+    for (let criterionIndex = 0; criterionIndex < MAX_CRITERIA_PER_CATALOG; criterionIndex++) {
+        criteria[criterionIndex] = {
+            field: 'enabled',
+            operator: Operator.EQUALS,
+            value: true,
+        };
+    }
+
+    const onChange = jest.fn();
+
+    render(
+        <ThemeProvider theme={pimTheme}>
+            <QueryClientProvider client={new QueryClient()}>
+                <ProductSelection criteria={criteria} onChange={onChange} errors={{}} />
+            </QueryClientProvider>
+        </ThemeProvider>
+    );
```

Comment:
```suggestion
    render(
        <ThemeProvider theme={pimTheme}>
            <QueryClientProvider client={new QueryClient()}>
                <ProductSelection criteria={criteria} onChange={jest.fn()} errors={{}} />
            </QueryClientProvider>
        </ThemeProvider>
    );
```

no need to have onChange in a variable, you don't assert on it :)